crates/wit-parser/src/lib.rs

```diff
@@ -17,6 +17,7 @@ pub mod abi;
 mod ast;
 use ast::lex::Span;
 pub use ast::SourceMap;
+pub use ast::{parse_use_path, ParsedUsePath};
```

Comment:
My reasoning was that a `use` path and a "fully-qualified" name, used for identifying something in a WIT file that we are interested for whatever reason, are very similar. For example, there are reasons to want to be able to parse all of the following examples in contexts outside of WIT source files:
```
foo:name/my-cool-world
foo:name/my-cool-world@1.2.3
foo:name/my-cool-world@1.2.3-rc-2023-11-10
foo:name/my-cool-world@1.2.3-rc-2023-11-10.{my-cool-type}
```

My particular use case was for reading `--world [MAYBE_FULLY_QUALIFIED_WORLD_NAME]` from the command line.

I could imagine that this is the a premature abstraction though, since it would also be able to read something like the following, which is probably a bit excessive:
```
foo:name/my-cool-world@1.2.3-rc-2023-11-10.{my-cool-type as my-cool-alias}
```

It's also possible there's another library function for parsing identifiers in the general sense that I missed?

Reply:
Nah makes sense to me! If you've got a need for it I think that it's fine to expose.

That being said for common translations my hope is that it'd go through helpers like `select_world` so most users wouldn't need to reach for this. That doesn't work for everyone though so seems reasonable to have this.

Reply:
Okay, I'll leave as is for now.